enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/BaseHighLimitRecordFormat.java

```diff
@@ -160,6 +160,9 @@ public void read( RECORD record, PageCursor primaryCursor, RecordLoad mode, int
             record.setUseFixedReferences( isUseFixedReferences( headerByte ) );
             doReadInternal( record, primaryCursor, recordSize, headerByte, inUse );
         }
+
+        // Set cursor offset to next record to prepare next read in case of scanning.
+        primaryCursor.setOffset( primaryStartOffset + recordSize );
```

Comment:
i not really sure why we set cursor offset inside of read here? and not outside, can you please give more wide description of why this needs to be done here and also why in all versions of BaseHighLimitRecordFormat?

Reply:
It's because this optimization relies on every entity record format to leave the cursor offset positioned on the first byte in the next record, after a read. It's done by the read method because this happens anyway during regular reads, so no extra cost is introduces. When std format records are not in use we need to explicitly advance the offset, and for all high-limit formats, because there we never know how much we've read because of the dynamic encoding.

I should probably add a test to ensure this offset behavior... will do that.

Reply:
See if you can make the test parameterised across formats, and for all stores if possible.

Reply:
Added it to `AbstractRecordFormatTest`, see last commit. That seems to do all current formats for nodes + relationships. Good enough?

Reply:
Not really, it's just that we only scan nodes and relationships, so all other formats can read as they please. I also haven't modified any other format read code.